.github/workflows/rust.yml

```diff
@@ -179,6 +179,59 @@ jobs:
                     --arg ddtags "$tags" \
                     '{ message: $message, ddsource: "hash-gh-actions", hostname: "github", service: "actions", ddtags: $ddtags }')
 
+  coverage:
+    name: coverage
+    needs: setup
+    if: needs.setup.outputs.coverage != '{}'
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix: ${{ fromJSON(needs.setup.outputs.coverage) }}
+    steps:
+      - name: Checkout source code
+        uses: actions/checkout@v3
+
+      - name: Install Rust
+        uses: actions-rs/toolchain@v1
+        with:
+          profile: minimal
+          toolchain: ${{ matrix.toolchain }}
+          override: true
+
+      - name: Cache Node dependencies
+        uses: actions/cache@v3
+        with:
+          path: |
+            **/node_modules
+          key: ${{ hashFiles('**/yarn.lock') }}
+
+      - name: Cache Rust dependencies
+        uses: Swatinem/rust-cache@v2
+        with:
+          workspaces: ${{ matrix.directory }}
+          key: ${{ matrix.profile }}
+
+      - name: Install tools
+        shell: bash
+        run: |
+          cargo install cargo-quickinstall
+          cargo quickinstall cargo-make --version 0.35.15
+          cargo quickinstall cargo-nextest --version 0.9.28
+          cargo quickinstall cargo-llvm-cov --version 0.5.0
+
+      - name: Run tests
+        working-directory: ${{ matrix.directory }}
+        run: cargo +${{ matrix.toolchain }} make coverage --lcov --output-path lcov.info
```

Comment:
I wonder if it’s necessary to run tests again as a separate job. Not sure how easy it is in Rust, but in JS projects I’ve seen folks passing an env var to the matix. E.g.:

Node.js A: `yarn test`
Node.js B (LTS): `TEST_COVERAGE=true yarn test`
Node.js C: `yarn test`
Node.js D: `yarn test`

WDYT?

Reply:
I plan to merge coverage and test jobs as soon as we have carried forward flags so we don't need to run coverage unconditionally